Bugsnag/Delivery/BugsnagApiClient.m

```diff
@@ -91,16 +91,19 @@ - (void)sendJSONPayload:(NSDictionary *)payload
     mutableHeaders[BugsnagHTTPHeaderNameIntegrity] = [NSString stringWithFormat:@"sha1 %@", [self SHA1HashStringWithData:data]];
     
     NSMutableURLRequest *request = [self prepareRequest:url headers:mutableHeaders];
+    bsg_log_debug(@"Sending %lu byte payload to %@", (unsigned long)data.length, url);
     
     [[self.session uploadTaskWithRequest:request fromData:data completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
         if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+            bsg_log_debug(@"Request to %@ completed with error %@", url, error);
             return completionHandler(BugsnagApiClientDeliveryStatusFailed, error ?:
                                      [NSError errorWithDomain:@"BugsnagApiClientErrorDomain" code:0 userInfo:@{
                                          NSLocalizedDescriptionKey: @"Request failed: no response was received",
                                          NSURLErrorFailingURLErrorKey: url }]);
         }
         
         NSInteger statusCode = ((NSHTTPURLResponse *)response).statusCode;
+        bsg_log_debug(@"Request to %@ completed with status code %ld", url, (long)statusCode);
```

Comment:
Could we log the response body if the code is not 2xx?

Reply:
Added in e9597208edb15243160ca25e1105435ca2c0eb52